### PR TITLE
Add env overloads for InclusiveSum and future ExclusiveScan

### PR DESCRIPTION
**Pull request by gonidelis**
Adds miscellaneous env overloads for `InclusiveSum` and `ExclusiveScan` with `FutureValue`

### CODE REVIEW

File: cub/cub/device/device_scan.cuh [hunk 1]
```diff
@@ -1068,6 +1068,102 @@ struct DeviceScan
...
+  //!
+  //! - Supports non-commutative scan operators.
+  //! - Results are not deterministic for pseudo-associative operators (e.g.,
+  //!   addition of floating-point types). Results for pseudo-associative
+  //!   operators may vary from run to run. Additional details can be found in
+  //!   the @lookback description.
+  //! - When ``d_in`` and ``d_out`` are equal, the scan is performed in-place. The
+  //!   range ``[d_in, d_in + num_items)`` and ``[d_out, d_out + num_items)``
+  //!   shall not overlap in any other way.
+  //!
```
Comment (gonidelis) [inline cub/cub/device/device_scan.cuh]: add env overload documentation info

File: cub/cub/device/device_scan.cuh [hunk 2]
```diff
@@ -1068,6 +1068,102 @@ struct DeviceScan
     return ExclusiveScan(d_temp_storage, temp_storage_bytes, d_data, d_data, scan_op, init_value, num_items, stream);
   }
 
+  //! @rst
+  //! Computes a device-wide exclusive prefix scan using the specified binary associative ``scan_op`` functor.
```
Comment (gonidelis) [inline cub/cub/device/device_scan.cuh:1073]: i am missing the in-place overloads

File: cub/test/catch2_test_device_scan_env.cu [hunk 3]
```diff
@@ -206,6 +207,24 @@ C2H_TEST("Device scan exclusive-sum can be tuned", "[scan][device]", block_sizes
...
+
+  REQUIRE(cudaSuccess == cub::DeviceScan::InclusiveSum(d_in, d_out.begin(), num_items));
+
+  for (int i = 0; i < num_items; i++)
+  {
+    REQUIRE(d_out[i] == (i + 1));
+  }
+}
+
 TEST_CASE("Device scan inclusive-scan works with default environment", "[scan][device]")
```
Comment (gonidelis) [inline cub/test/catch2_test_device_scan_env.cu:205]: exclusive scan future is missing default environment test

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 4]
```diff
@@ -110,6 +110,99 @@ C2H_TEST("cub::DeviceScan::ExclusiveSum accepts stream and not_guaranteed determ
...
+  {
+    std::cerr << "cub::DeviceScan::ExclusiveScan (FutureValue) failed with status: " << error << std::endl;
+  }
+
+  thrust::device_vector<int> expected{10, 11, 13, 16};
+
+  REQUIRE(error == cudaSuccess);
+  REQUIRE(output == expected);
+}
+
```
Comment (gonidelis) [inline cub/test/catch2_test_device_scan_env_api.cu:213]: add a exclusive scan accepting `non_guaranteed` for uniformity

File: cub/cub/device/device_scan.cuh (hunk 2 above)
Comment (gonidelis) [reply cub/cub/device/device_scan.cuh:1073]: https://github.com/NVIDIA/cccl/issues/8060

File: cub/cub/device/device_scan.cuh [hunk 5]
```diff
@@ -328,10 +328,6 @@ struct DeviceScan
   //!   addition of floating-point types). Results for pseudo-associative
   //!   operators may vary from run to run. Additional details can be found in
   //!   the @lookback description.
-  //! - When ``d_in`` and ``d_out`` are equal, the scan is performed in-place.
-  //!   The range ``[d_in, d_in + num_items)`` and ``[d_out, d_out + num_items)``
-  //!   shall not overlap in any other way.
```
Comment (bernhardmgruber) [inline cub/cub/device/device_scan.cuh:321]: Q: Why are we removing this? I think this documentation still applies.

File: cub/cub/device/device_scan.cuh [hunk 6]
```diff
@@ -1068,6 +1063,99 @@ struct DeviceScan
...
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    ScanOpT scan_op,
+    FutureValue<InitValueT, InitValueIterT> init_value,
+    NumItemsT num_items,
+    EnvT env = {})
+  {
+    _CCCL_NVTX_RANGE_SCOPE("cub::DeviceScan::ExclusiveScan");
+
+    return scan_impl_env(d_in, d_out, scan_op, detail::InputValue<InitValueT>(init_value), num_items, env);
```
Comment (bernhardmgruber) [inline cub/cub/device/device_scan.cuh:1162]: Remark: The non-env version has a `using OffsetT = detail::choose_offset_t<NumItemsT>;` which I don't see here nor in `scan_impl_env`. It's finally in `scan_impl_determinism`, so we should be good here.

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 7]
```diff
@@ -39,6 +39,27 @@ C2H_TEST("cub::DeviceScan::ExclusiveScan accepts run_to_run determinism requirem
...
+  if (error != cudaSuccess)
+  {
+    std::cerr << "cub::DeviceScan::ExclusiveScan failed with status: " << error << std::endl;
+  }
+
+  thrust::device_vector<int> expected{0, 0, 1, 3};
+
+  REQUIRE(error == cudaSuccess);
+  REQUIRE(output == expected);
+}
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env_api.cu]: Important: This example seems unused by any documentation. Please add it.

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 8]
```diff
@@ -59,6 +80,7 @@ C2H_TEST("cub::DeviceScan::ExclusiveScan accepts stream and not_guaranteed deter
   }
 
   thrust::device_vector<float> expected{1.0f, 1.0f, 2.0f, 4.0f};
+  stream.sync();
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env_api.cu]: Q: Why do we need to show a sync in the documentation here? It's not necessary IMO.

Applies again below.

File: cub/cub/device/device_scan.cuh [hunk 9]
```diff
@@ -334,7 +334,6 @@ struct DeviceScan
   //! - When ``d_in`` and ``d_out`` are equal, the scan is performed in-place.
   //!   The range ``[d_in, d_in + num_items)`` and ``[d_out, d_out + num_items)``
   //!   shall not overlap in any other way.
-  //! - @devicestorage
```
Comment (bernhardmgruber) [inline cub/cub/device/device_scan.cuh]: I was first confused that we remove `@devicestorage` for an old overload. But now I see it's an env-overload. But then this cannot be true:
```c++
  //! .. versionadded:: 2.2.0
  //!    First appears in CUDA Toolkit 12.3.
```
Please fix the versions.

File: cub/cub/device/device_scan.cuh [hunk 10]
```diff
@@ -1071,6 +1078,100 @@ struct DeviceScan
     return ExclusiveScan(d_temp_storage, temp_storage_bytes, d_data, d_data, scan_op, init_value, num_items, stream);
   }
 
+  //! @rst
+  //! Computes a device-wide exclusive prefix scan using the specified binary associative ``scan_op`` functor.
```
Comment (bernhardmgruber) [inline cub/cub/device/device_scan.cuh:1073]: The newly added overloads don't have these `versionadded` documentation. Please add it.

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 7 above)
Comment (bernhardmgruber) [reply cub/test/catch2_test_device_scan_env_api.cu]: @gonidelis you resolved this comment but there are still no `// example-begin` etc. comments in this example. Therefore, it's still not used anywhere. Please address this.

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 11]
```diff
@@ -60,6 +81,7 @@ C2H_TEST("cub::DeviceScan::ExclusiveScan accepts stream and not_guaranteed deter
 
   thrust::device_vector<float> expected{1.0f, 1.0f, 2.0f, 4.0f};
   // example-end exclusive-scan-env-stream
+  stream.sync();
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env_api.cu:63]: Q: Why do we need a sync here? I don't think we need it.

Applies more often in this file.

File: cub/test/catch2_test_device_scan_env.cu [hunk 12]
```diff
@@ -331,6 +399,29 @@ C2H_TEST("Device scan exclusive-sum uses environment", "[scan][device]")
...
+          == cub::DeviceScan::InclusiveSum(nullptr, expected_bytes_allocated, d_in.begin(), d_out.begin(), num_items));
+
+  auto env = stdexec::env{cuda::execution::require(cuda::execution::determinism::not_guaranteed), // determinism
+                          expected_allocation_size(expected_bytes_allocated)}; // temp storage size
+
+  device_scan_inclusive_sum(d_in.begin(), d_out.begin(), num_items, env);
+
+  for (int i = 0; i < num_items; i++)
+  {
+    REQUIRE(d_out[i] == (i + 1));
```
Comment (pauleonix) [inline cub/test/catch2_test_device_scan_env.cu]: Suggestion: Avoid looping over `device_vector`, use helper/`thrust::equal` instead to avoid unnecessary slowdown.

File: cub/test/catch2_test_device_scan_env.cu [hunk 13]
```diff
@@ -331,6 +399,29 @@ C2H_TEST("Device scan exclusive-sum uses environment", "[scan][device]")
   }
 }
 
+C2H_TEST("Device scan inclusive-sum uses environment", "[scan][device]")
+{
+  using num_items_t = int;
+
+  num_items_t num_items = 10;
```
Comment (pauleonix) [inline cub/test/catch2_test_device_scan_env.cu]: Use (small) random value to include corner-cases.

File: cub/test/catch2_test_device_scan_env.cu [hunk 14]
```diff
@@ -307,6 +347,34 @@ C2H_TEST("Device scan exclusive-scan uses environment", "[scan][device]")
...
+C2H_TEST("Device scan exclusive-scan with FutureValue uses environment", "[scan][device]")
+{
+  using scan_op_t   = cuda::std::plus<>;
+  using num_items_t = int;
+
+  num_items_t num_items = 10;
+  auto d_in             = cuda::constant_iterator(1);
+  auto d_out            = thrust::device_vector<int>(num_items);
+
+  auto init_value_vec = thrust::device_vector<int>{0};
```
Comment (pauleonix) [inline cub/test/catch2_test_device_scan_env.cu]: Should not be the neutral element of the binary operator, so `FutureValue` makes a difference.

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 15]
```diff
@@ -39,6 +39,27 @@ C2H_TEST("cub::DeviceScan::ExclusiveScan accepts run_to_run determinism requirem
   REQUIRE(output == expected);
 }
 
+C2H_TEST("cub::DeviceScan::ExclusiveScan accepts not_guaranteed determinism requirements", "[scan][env]")
+{
+  auto op     = cuda::std::plus{};
+  auto input  = thrust::device_vector<int>{0, 1, 2, 3};
+  auto output = thrust::device_vector<int>(4);
+  auto init   = 0;
```
Comment (pauleonix) [inline cub/test/catch2_test_device_scan_env_api.cu]: Same issue with the initial value being the neutral element for the given binary op.

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 16]
```diff
@@ -39,6 +39,27 @@ C2H_TEST("cub::DeviceScan::ExclusiveScan accepts run_to_run determinism requirem
   REQUIRE(output == expected);
 }
 
+C2H_TEST("cub::DeviceScan::ExclusiveScan accepts not_guaranteed determinism requirements", "[scan][env]")
+{
+  auto op     = cuda::std::plus{};
+  auto input  = thrust::device_vector<int>{0, 1, 2, 3};
```
Comment (pauleonix) [inline cub/test/catch2_test_device_scan_env_api.cu]: This could be a counting iter with a random number of values.

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 11 above)
Comment (pauleonix) [reply cub/test/catch2_test_device_scan_env_api.cu:63]: Initializing `expected` implicitly syncs, but it is good to be explicit about it.

File: cub/test/catch2_test_device_scan_env.cu (hunk 13 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env.cu]: this would require to make the logic for the `expected` vectors more complex, do you think this change is necessary? are you referring to zero sized inputs type of corner cases?

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 7 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env_api.cu]: there might be api test cases where we don't insert in the docs in order to avoid bloating. i added the guards but did not insert it in the rst docs with a literalinclude as a non_guaranteed example already exists

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 11 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env_api.cu:63]: i added for me to remove this discrepancy here. afaiu we should be using `stream.sync()` but now show it to the users as it should be left on their judgement for when to use it

https://github.com/NVIDIA/cccl/issues/8175#issue-4138038297

File: cub/test/catch2_test_device_scan_env_api.cu [hunk 17]
```diff
@@ -105,6 +129,106 @@ C2H_TEST("cub::DeviceScan::ExclusiveSum accepts stream and not_guaranteed determ
...
+
+  auto error = cub::DeviceScan::InclusiveSum(input.begin(), output.begin(), input.size(), env);
+  if (error != cudaSuccess)
+  {
+    std::cerr << "cub::DeviceScan::InclusiveSum failed with status: " << error << std::endl;
+  }
+
+  thrust::device_vector<float> expected{1.0f, 3.0f, 6.0f, 10.0f};
+  // example-end inclusive-sum-env-stream
+  stream.sync();
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env_api.cu:155]: Important: we should not need any `sync` here. Also not in any other tests in this file. Why is it needed?

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 11 above)
Comment (bernhardmgruber) [reply cub/test/catch2_test_device_scan_env_api.cu:63]: > Initializing `expected` implicitly syncs

But the code here does not rely on this, so it's fine if it wouldn't.

> but it is good to be explicit about it.

I disagree. No sync is needed here, so seeing one makes readers like me nervous.

File: cub/test/catch2_test_device_scan_env.cu [hunk 18]
```diff
@@ -80,6 +82,25 @@ TEST_CASE("Device scan exclusive scan works with default environment", "[scan][d
   REQUIRE(d_block_size[0] == target_block_size);
 }
 
+TEST_CASE("Device scan exclusive scan with FutureValue works with default environment", "[scan][device]")
```
Comment (gonidelis) [inline cub/test/catch2_test_device_scan_env.cu:85]: Add failed tests for `float` `run_to_run` here:
/home/ggonidelis/cccl/cub/test/test_device_exclusive_scan_determinism_fail.cu

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 7 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env_api.cu]: we can also remove that

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 8 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env_api.cu]: Do add `stream.sync()`s but add them outside of the `literalinclude` per Bernhard's suggestion

File: cub/test/catch2_test_device_scan_env.cu (hunk 13 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env.cu]: Do use `GENERATE()` to generate size of input but elements are still all `1` using constant_iter

File: cub/cub/device/device_scan.cuh [hunk 19]
```diff
@@ -334,15 +334,6 @@ struct DeviceScan
   //! - When ``d_in`` and ``d_out`` are equal, the scan is performed in-place.
   //!   The range ``[d_in, d_in + num_items)`` and ``[d_out, d_out + num_items)``
   //!   shall not overlap in any other way.
-  //! - @devicestorage
-  //!
-  //! Preconditions
-  //! +++++++++++++
-  //!
-  //! - When ``d_in`` and ``d_out`` are equal, the scan is performed in-place.
```
Comment (pauleonix) [inline cub/cub/device/device_scan.cuh:327]: Q: Is the removal of the preconditions deliberate?

Comment (pauleonix) [reply cub/cub/device/device_scan.cuh:327]: I see that the first precondition is already mentioned above.

File: cub/cub/device/device_scan.cuh [hunk 20]
```diff
@@ -1071,6 +1070,103 @@ struct DeviceScan
...
+  //!    First appears in CUDA Toolkit 13.4.
+  //!
+  //! - Can use a specific stream or cuda memory resource through the ``env`` parameter.
+  //! - Supports non-commutative scan operators.
+  //! - Results are not deterministic for pseudo-associative operators (e.g.,
+  //!   addition of floating-point types). Results for pseudo-associative
+  //!   operators may vary from run to run. Additional details can be found in
+  //!   the @lookback description.
+  //!
+  //! Preconditions
```
Comment (pauleonix) [inline cub/cub/device/device_scan.cuh]: Suggestion: A similar section was removed above. If this one is removed as well, the first precondition needs to be added back to the preceding list of notes.

File: cub/cub/device/device_scan.cuh [hunk 21]
```diff
@@ -1071,6 +1070,103 @@ struct DeviceScan
...
+  //! @tparam OutputIteratorT
+  //!   **[inferred]** Random-access output iterator type for writing scan outputs @iterator
+  //!
+  //! @tparam ScanOpT
+  //!   **[inferred]** Binary associative scan functor type having member `T operator()(const T &a, const T &b)`
+  //!
+  //! @tparam InitValueT
+  //!  **[inferred]** Type of the `init_value`
+  //!
+  //! @tparam InitValueIterT
```
Comment (pauleonix) [inline cub/cub/device/device_scan.cuh:1115]: Suggestion: Documentation of this template parameter is missing from the non-env member functions (adding it to them is out-of-scope for this PR). I wonder if the documentation could be a bit more specific. Currently it seems to give no more information than the name of the template parameter.

File: cub/cub/device/device_scan.cuh [hunk 22]
```diff
@@ -1260,6 +1356,76 @@ struct DeviceScan
...
+  //! .. versionadded:: 3.4.0
+  //!    First appears in CUDA Toolkit 13.4.
+  //!
+  //! - Supports non-commutative sum operators.
+  //! - Results are not deterministic for pseudo-associative operators (e.g.,
+  //!   addition of floating-point types). Results for pseudo-associative
+  //!   operators may vary from run to run. Additional details can be found in
+  //!   the @lookback description.
+  //!
+  //! Preconditions
```
Comment (pauleonix) [inline cub/cub/device/device_scan.cuh]: Suggestion: Another added "Preconditions" section.

File: cub/test/catch2_test_device_scan_env.cu (hunk 13 above)
Comment (pauleonix) [reply cub/test/catch2_test_device_scan_env.cu]: @gevtushenko brought this up (as a nit I assume). I don't see why the logic would be more complex. The idea from your second comment seems fine to me.

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 7 above)
Comment (bernhardmgruber) [reply cub/test/catch2_test_device_scan_env_api.cu]: Please either use the API example in the docs, or remove it. I am fine with both!

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 17 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env_api.cu:155]: in case llms catch that we do need the sync eventually because running stream is not default and == happens in the default stream

File: cub/test/catch2_test_device_scan_env.cu (hunk 18 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env.cu:85]: done

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 7 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env_api.cu]: removed

File: cub/test/catch2_test_device_scan_env.cu [hunk 23]
```diff
@@ -80,6 +82,25 @@ TEST_CASE("Device scan exclusive scan works with default environment", "[scan][d
...
+{
+  using num_items_t = int;
+
+  num_items_t num_items = 4;
+
+  auto d_in  = thrust::device_vector<int>{1, 1, 1, 1};
+  auto d_out = thrust::device_vector<int>(num_items);
+
+  auto init_value_vec = thrust::device_vector<int>{42};
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env.cu]: Please use the C2H vectors, since they work better on Tegra platforms (handle out of memory more gracefully)
```suggestion
  auto d_in  = c2h::device_vector<int>{1, 1, 1, 1};
  auto d_out = c2h::device_vector<int>(num_items);

  auto init_value_vec = c2h::device_vector<int>{42};
```
Applies everywhere in CUB tests (except API tests that we put in the documentation)

File: cub/test/catch2_test_device_scan_env.cu [hunk 24]
```diff
@@ -80,6 +82,25 @@ TEST_CASE("Device scan exclusive scan works with default environment", "[scan][d
...
+  auto future_init    = cub::FutureValue<int>(thrust::raw_pointer_cast(init_value_vec.data()));
+
+  REQUIRE(cudaSuccess
+          == cub::DeviceScan::ExclusiveScan(d_in.begin(), d_out.begin(), cuda::std::plus{}, future_init, num_items));
+
+  auto expected = thrust::device_vector<int>{42, 43, 44, 45};
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env.cu]: Same here

File: cub/test/catch2_test_device_scan_env.cu [hunk 25]
```diff
@@ -80,6 +82,25 @@ TEST_CASE("Device scan exclusive scan works with default environment", "[scan][d
...
+  REQUIRE(thrust::equal(d_out.begin(), d_out.end(), expected.begin()));
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env.cu]: Why doesn't this work?
```suggestion
  REQUIRE(expected == d_out);
```

File: cub/test/catch2_test_device_scan_env.cu [hunk 26]
```diff
@@ -294,7 +356,33 @@ C2H_TEST("Device scan exclusive-sum uses environment", "[scan][device]")
...
+  auto env = stdexec::env{cuda::execution::require(cuda::execution::determinism::not_guaranteed), // determinism
+                          expected_allocation_size(expected_bytes_allocated)}; // temp storage size
+
+  device_scan_inclusive_sum(d_in.begin(), d_out.begin(), num_items, env);
+
+  thrust::host_vector<int> h_expected(num_items);
+  for (int i = 0; i < num_items; i++)
+  {
+    h_expected[i] = 3 * (i + 1);
+  }
```
Comment (bernhardmgruber) [inline cub/test/catch2_test_device_scan_env.cu:383]:
```suggestion
  thrust::host_vector<int> h_expected{3, 6, 9};
```

File: cub/test/catch2_test_device_scan_env_api.cu (hunk 11 above)
Comment (bernhardmgruber) [reply cub/test/catch2_test_device_scan_env_api.cu:63]: I discussed this with @gonidelis and I now understand why we need the sync. We wondered why the code still works without the sync. Where does

> Initializing expected implicitly syncs

Is it the memory allocation? I just want to understand why it works without a sync.

We should definitely add the syncs now!

File: cub/test/catch2_test_device_scan_env.cu (hunk 23 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env.cu]: bad slip up thanks for catching

File: cub/test/catch2_test_device_scan_env.cu (hunk 26 above)
Comment (gonidelis) [reply cub/test/catch2_test_device_scan_env.cu:383]: `num_items` is variable after team review on Friday. 

`  num_items_t num_items = GENERATE(0, 1, 10);`

Comment (bernhardmgruber) [reply cub/test/catch2_test_device_scan_env.cu:383]: Ah sorry, fine then!

**Comment by bernhardmgruber** [comment]
There are errors in the docs build:
```
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:348: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:654: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:1123: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:1393: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:1782: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:1867: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:2603: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:2732: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:2860: error: explicit link request to 'cuda::std::execution::env' could not be resolved
/home/runner/_work/cccl/cccl/cub/cub/device/device_scan.cuh:2981: error: explicit link request to 'cuda::std::execution::env' could not be resolved
```
Please address those.
